src/result/mod.rs

```diff
@@ -47,7 +47,7 @@ use std::fmt::{Display, Formatter, Result as FmtResult};
 ///
 /// [unit]: https://doc.rust-lang.org/book/ch05-01-defining-structs.html#unit-like-structs-without-any-fields
 #[derive(Debug)]
-pub struct Throw;
+pub struct Throw(pub(crate) ());
```

Comment:
What do you think about using a `!Send + !Sync` type here to make it a _little_ more difficult? Also, I would recommend a private constructor.

```rust
pub struct Throw(PhantomData<*mut c_void>);

impl Throw {
    pub(crate) fn new() -> Self {
        Self(PhantomData)
    }
}
```